pageserver/src/tenant/mgr.rs

```diff
@@ -413,6 +436,15 @@ pub async fn detach_tenant(
     conf: &'static PageServerConf,
     tenant_id: TenantId,
     detach_ignored: bool,
+) -> Result<(), TenantStateError> {
+    detach_tenant0(conf, &TENANTS, tenant_id, detach_ignored).await
+}
+
+async fn detach_tenant0(
+    conf: &'static PageServerConf,
+    tenants: &tokio::sync::RwLock<TenantsMap>,
+    tenant_id: TenantId,
```

Comment:
IMO it's better to use `_impl` suffix then `0`.
It avoids potential confusion about a "special tenant 0" which we may add in the future ("system tenant" / "catalog tenant").

There's are a few other functions that also had this, please adjust it as well.

Reply:
I disagree. How could anyone think that we have some system tenant if you see a public method and a private one, the other is oneline and the other is the real implementation.

Reply:
Yeah, admittedly a weak argument.

The other argument I have is that we already have a bunch of `_impl` functions (yeah, it's been mostly me who introduced them).